src/TaskRunner.js

```diff
@@ -40,8 +40,26 @@ export default class TaskRunner {
       if (this.worker.getStderr()) this.worker.getStderr().pipe(process.stderr);
     }
 
-    return Promise.all(
-      tasks.map((task) => {
+    let inputIndex = -1;
+    let outputIndex = 0;
+    const handlers = {};
+
+    const offerResult = (idx, task, completedTask) => {
+      return new Promise((resolve) => {
+        handlers[idx] = () => {
+          onCompletedTask(task, completedTask);
+          delete handlers[idx];
+          resolve();
+        };
+        while (outputIndex in handlers) {
+          handlers[outputIndex]();
+          outputIndex += 1;
+        }
+      });
+    };
```

Comment:
Looks like it is kill idea of parallels, because you run next task after other task